### PR TITLE
UAR-397: Fix feature flag logic on save and continue button

### PR DESCRIPTION
**Pull request by sjahir**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-397

### Change description
- Display “Save and Continue” or “Continue” for submit buttons depending on the feature flags for register and update journey

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### CODE REVIEW

File: test/controllers/update/overseas.entity.presenter.controller.spec.ts [hunk 1]
```diff
@@ -76,6 +77,7 @@ describe("OVERSEAS ENTITY PRESENTER controller", () => {
       expect(resp.text).toContain(OVERSEAS_ENTITY_PRESENTER_PAGE_TITLE);
       expect(resp.text).not.toContain(PAGE_TITLE_ERROR);
       expect(resp.text).toContain(UPDATE_USE_INFORMATION_NEED_MORE);
+      expect(resp.text).toContain(CONTINUE_BUTTON_TEXT);
```
Comment (AndreKramer7654) [inline test/controllers/update/overseas.entity.presenter.controller.spec.ts:80]: if (!config.FEATURE_ENABLED_UPDATE_SAVE_AND_RESUME) {
expect(resp.text).toContain(CONTINUE_BUTTON_TEXT); } else
{ expect(resp.text).toContain(SAVE_AND_CONTINUE_BUTTON_TEXT); }

File: test/controllers/update/update.beneficial.owner.gov.spec.ts [hunk 2]
```diff
@@ -96,7 +96,7 @@ describe("UPDATE BENEFICIAL OWNER GOV controller", () => {
       expect(resp.text).toContain(LANDING_PAGE_URL);
       expect(resp.text).not.toContain(PAGE_TITLE_ERROR);
       expect(resp.text).toContain(BENEFICIAL_OWNER_GOV_PAGE_HEADING);
-      expect(resp.text).toContain(SAVE_AND_CONTINUE_BUTTON_TEXT);
+      expect(resp.text).toContain(CONTINUE_BUTTON_TEXT);
```
Comment (AndreKramer7654) [inline test/controllers/update/update.beneficial.owner.gov.spec.ts:99]: As above

File: test/controllers/update/update.beneficial.owner.gov.spec.ts [hunk 3]
```diff
@@ -124,7 +124,7 @@ describe("UPDATE BENEFICIAL OWNER GOV controller", () => {
       expect(resp.text).toContain("LegalForm");
       expect(resp.text).toContain("a11");
       expect(resp.text).toContain("name=\"is_on_sanctions_list\" type=\"radio\" value=\"1\" checked");
-      expect(resp.text).toContain(SAVE_AND_CONTINUE_BUTTON_TEXT);
+      expect(resp.text).toContain(CONTINUE_BUTTON_TEXT);
```
Comment (AndreKramer7654) [inline test/controllers/update/update.beneficial.owner.gov.spec.ts:127]: As above

File: test/controllers/update/update.beneficial.owner.gov.spec.ts [hunk 4]
```diff
@@ -79,6 +87,8 @@ const mockSetApplicationData = setApplicationData as jest.Mock;
 const mockMapFieldsToDataObject = mapFieldsToDataObject as jest.Mock;
 const DUMMY_DATA_OBJECT = { dummy: "data" };
 
+const isSaveAndContinueButton = FEATURE_FLAG_ENABLE_UPDATE_SAVE_AND_RESUME === "true" ? SAVE_AND_CONTINUE_BUTTON_TEXT : CONTINUE_BUTTON_TEXT;
```
Comment (AndreKramer7654) [inline test/controllers/update/update.beneficial.owner.gov.spec.ts:90]: please rename saveAndContuneButtonText

File: test/controllers/update/update.filing.date.controller.spec.ts [hunk 5]
```diff
@@ -67,6 +69,8 @@ mockServiceAvailabilityMiddleware.mockImplementation((req: Request, res: Respons
 const mockIsActiveFeature = isActiveFeature as jest.Mock;
 mockIsActiveFeature.mockReturnValue(true);
 
+const isSaveAndContinueButton = config.FEATURE_FLAG_ENABLE_UPDATE_SAVE_AND_RESUME === "true" ? SAVE_AND_CONTINUE_BUTTON_TEXT : CONTINUE_BUTTON_TEXT;
```
Comment (AndreKramer7654) [inline test/controllers/update/update.filing.date.controller.spec.ts:72]: please share

**Review by AndreKramer7654 — APPROVED** [review]

**Review by azafarCH — APPROVED** [review]

**Review by bantao-ch — APPROVED** [review]

**Review by jkayyala — APPROVED** [review]

**Review by mmcpolin-ch — APPROVED** [review]

File: views/includes/save-and-continue-button.html [hunk 6]
```diff
@@ -1,7 +1,13 @@
 {% set textContinueButton = "Continue" %}
 
-{% if OE_CONFIGS.FEATURE_FLAG_ENABLE_SAVE_AND_RESUME_17102022 == "true" or OE_CONFIGS.FEATURE_FLAG_ENABLE_UPDATE_SAVE_AND_RESUME == "true" %}
-  {% set textContinueButton = "Save and continue" %}
+{% if ('/update-an-overseas-entity' in backLinkUrl) %}
```
Comment (mwestacott) [inline views/includes/save-and-continue-button.html:3]: Using the URL here is a little bit 'brittle'. I'd say it's acceptable as long as a test - or tests - start failing if the URL is changed [here](https://github.com/companieshouse/overseas-entities-web/blob/67b0c3ab3e5a1bc0f50cf7b732814ed9ba837fee/src/config/index.ts#L164) but not in this view (alerting developers that a change is needed here too).

Comment (sjahir) [reply views/includes/save-and-continue-button.html:3]: Thats a good point! `OE_CONFIGS.UPDATE_LANDING_URL` would be the better approach.

**Review by mwestacott — APPROVED** [review]

**Pull request merged**